opendcs-rest-api/src/main/java/org/opendcs/odcsapi/res/PlatformResources.java

```
@@ -33,113 +37,469 @@
+			return Response.status(HttpServletResponse.SC_OK).entity(statusListMap(dbIo, statuses)).build();
+		}
+		catch (DbIoException ex)
+		{
+			throw new DbException(String.format("Unable to retrieve platform status with ID: %s", netlistId), ex);
+		}
+		finally
+		{
+			dbIo.close();
+		}
+	}
+
+	static List<ApiPlatformStatus> statusListMap(DatabaseIO dbIo, List<PlatformStatus> statuses)
 	{
-		Logger.getLogger(ApiConstants.loggerName).fine("gePlatformStats, netlistid=" + netlistId);
-		try (DbInterface dbi = new DbInterface();
-			ApiPlatformDAO platformDAO = new ApiPlatformDAO(dbi))
+		if(statuses == null)
+		{
+			return new ArrayList<>();
+		}
+		List<ApiPlatformStatus> ret = new ArrayList<>();
+		for(PlatformStatus status : statuses)
 		{
-			return ApiHttpUtil.createResponse(platformDAO.getPlatformStatus(netlistId));
+			ApiPlatformStatus ps = new ApiPlatformStatus();
+			if(status.getPlatformId() != null)
+			{
+				ps.setPlatformId(status.getPlatformId().getValue());
+			}
+			else
+			{
+				ps.setPlatformId(DbKey.NullKey.getValue());
+			}
+			ps.setAnnotation(status.getAnnotation());
+			ps.setLastContact(status.getLastContactTime());
+			ps.setLastError(status.getLastErrorTime());
+			ps.setLastMessage(status.getLastMessageTime());
+			if (dbIo != null && !status.getLastScheduleEntryStatusId().isNull()
+					&& (status.getLastRoutingSpecName() == null || status.getLastRoutingSpecName().isEmpty()))
+			{
+				try (ScheduleEntryDAI dai = dbIo.makeScheduleEntryDAO())
+				{
+					ScheduleEntry scheduleEntry = dai.readScheduleEntry(status.getLastScheduleEntryStatusId());
+					long routingId = scheduleEntry.getRoutingSpecId().getValue();
+					RoutingSpec rs = new RoutingSpec();
+					rs.setId(DbKey.createDbKey(routingId));
+					dbIo.readRoutingSpec(rs);
+					ps.setRoutingSpecName(rs.getName());
+				}
+				catch (DatabaseException | DbIoException ex)
+				{
+					// Ignore
```

never ignore exceptions

---

Fixed